cvat/apps/engine/serializers.py

```diff
@@ -1360,76 +1423,192 @@ def update(self, instance: models.Task, validated_data: dict[str, Any]) -> model
+        ):
+            models.Segment.objects.filter(
+                id__in=updated_segment_chunks_batch
+            ).update(chunks_updated_date=updated_date)
+
+        # Update parent objects
+        db_task.touch()
+        if db_task.project:
+            db_task.project.touch()
+
+    def _update_frames_in_bulk(
+        self,
+        db_task: models.Task,
+        *,
+        bulk_context: _TaskValidationLayoutBulkUpdateContext,
+    ):
+        self._clear_annotations_on_frames(db_task, bulk_context.updated_honeypots)
+
+        # The django generated bulk_update() query is too slow, so we use bulk_create() instead
+        # NOTE: Silk doesn't show these queries in the list of queries
+        # for some reason, but they can be seen in the profile
+        models.Image.objects.bulk_create(
+            list(bulk_context.updated_honeypots.values()),
+            update_conflicts=True,
+            update_fields=['path', 'width', 'height', 'real_frame'],
+            unique_fields=[
+                # required for Postgres
+                # https://docs.djangoproject.com/en/4.2/ref/models/querysets/#bulk-create
+                'id'
+            ],
+            batch_size=1000,
+        )
+
+        # Update related images in 2 steps: remove all m2m for honeypots, then add (copy) new ones
+        # 1. remove
+        for updated_honeypots_batch in take_by(bulk_context.updated_honeypots, chunk_size=1000):
+            models.RelatedFile.images.through.objects.filter(
+                image_id__in=updated_honeypots_batch
+            ).delete()
+
+        # 2. batched add (copy), collect all the new records and insert
+        frame_provider = TaskFrameProvider(db_task)
+        validation_frame_uses_with_related_files: dict[int, list[int]] = {}
+        for honeypot_frame, db_honeypot in bulk_context.updated_honeypots.items():
+            validation_frame_uses_with_related_files.setdefault(
+                frame_provider.get_rel_frame_number(db_honeypot.real_frame), []
+            ).append(honeypot_frame)
+
+        new_m2m_objects = []
+        related_files_m2m_objects = (
+            models.RelatedFile.images.through.objects
+            .filter(image_id__in=[
+                bulk_context.all_db_frames[f].id for f in db_task.data.validation_layout.frames
+            ])
+            .all()
+        )
+        for validation_frame, validation_frame_uses in (
+            validation_frame_uses_with_related_files.items()
+        ):
+            new_m2m_objects.extend(
+                models.RelatedFile.images.through(
+                    image_id=bulk_context.all_db_frames[honeypot_frame].id,
+                    related_file_id=related_file_m2m_obj.related_file_id
+                )
+                for honeypot_frame in validation_frame_uses
+                for related_file_m2m_obj in related_files_m2m_objects
+                if related_file_m2m_obj.image_id == validation_frame
```

Comment:
Shouldn't there be an `id` of an `Image` object instead of the real frame id in a task?

Reply:
Thanks, seems to be an error.